### PR TITLE
Replaces crossing pointer with plain member variable

### PR DESCRIPTION
**Pull request by zeroset**
This Crossing belongs to a certain goal. No need for dynamic allocation
here.

### CODE REVIEW

File: geometry/Goal.cpp [hunk 1]
```diff
@@ -37,14 +37,12 @@ Goal::Goal()
...
      _walls = std::vector<Wall > ();
      _poly = std::vector<Point > ();
-     _crossing = new Crossing();
      _roomID = -1;
      _subRoomID = -1;
 }
 
 Goal::~Goal()
 {
-
```
Comment (schroedtert) [inline geometry/Goal.cpp]: What about using `~Goal() = default` in header?

File: geometry/Goal.cpp [hunk 2]
```diff
@@ -234,16 +232,16 @@ bool Goal::ConvertLineToPoly()
...
      }else{
-          _crossing->SetPoint1(_poly[0]);
+          _crossing.SetPoint1(_poly[0]);
           Line tmp_line(_poly[_poly.size()/2], _poly[(_poly.size()/2)+1], 0);
-          _crossing->SetPoint2(tmp_line.GetCentre());
+          _crossing.SetPoint2(tmp_line.GetCentre());
      }
 
 
-//     std::cout << "Crossing goal: " << _crossing->GetUniqueID() << _crossing->toString() << std::endl;
```
Comment (schroedtert) [inline geometry/Goal.cpp]: Can be deleted. Blame me for the comment.

File: geometry/Goal.cpp (hunk 1 above)
Comment (zeroset) [reply geometry/Goal.cpp]: Changed, thanks.

File: geometry/Goal.cpp (hunk 2 above)
Comment (zeroset) [reply geometry/Goal.cpp]: removed ;)

**Pull request merged**